src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/AssemblyAttributeProperties/AssemblyInfoProperties.cs

```diff
@@ -102,24 +100,37 @@ await SaveAssemblyInfoInSourceAsync().ConfigureAwait(true))
         public async override Task SetPropertyValueAsync(string propertyName, string unevaluatedPropertyValue, IReadOnlyDictionary<string, string> dimensionalConditions = null)
         {
             if (_attributeValueProviderMap.ContainsKey(propertyName) &&
-                await SaveAssemblyInfoInSourceAsync().ConfigureAwait(true))
+                await GenerateAssemblyInfoPropertyInSourceAsync(propertyName).ConfigureAwait(true))
             {
-                if (_attributeValueProviderMap.TryGetValue(propertyName, out SourceAssemblyAttributePropertyValueProvider provider))
-                {
-                    await provider.SetPropertyValueAsync(unevaluatedPropertyValue).ConfigureAwait(true);
-                }
+                SourceAssemblyAttributePropertyValueProvider provider = _attributeValueProviderMap[propertyName];
+                await provider.SetPropertyValueAsync(unevaluatedPropertyValue).ConfigureAwait(true);
             }
             else
             {
                 await base.SetPropertyValueAsync(propertyName, unevaluatedPropertyValue, dimensionalConditions).ConfigureAwait(true);
             }
         }
 
-        private async Task<bool> SaveAssemblyInfoInSourceAsync()
+        private async Task<bool> GenerateAssemblyInfoPropertyInSourceAsync(string propertyName)
```

Comment:
The method name is confusing. I thought this method generates the attribute. Maybe rename to IsAssemblyInfoPropertyGeneratedByBuild() and reverse the semantics? Or rename to IsAssemblyInfoPropertyDeclaredInSource()?